sunpy/net/dataretriever/sources/goes.py

```diff
@@ -269,12 +269,18 @@ def search(self, *args, **kwargs):
         matchdict = self._get_match_dict(*args, **kwargs)
         req_wave = matchdict.get('Wavelength', None)
         if req_wave is not None:
-            wmin = req_wave.min.to(u.Angstrom, equivalencies=u.spectral())
-            wmax = req_wave.max.to(u.Angstrom, equivalencies=u.spectral())
-            req_wave = a.Wavelength(wmin, wmax)
+            # wmin = (req_wave.min.to(u.Angstrom, equivalencies=u.spectral()))
+            # wmax = (req_wave.max.to(u.Angstrom, equivalencies=u.spectral()))
+
+            #casting it to int
+            wmin_round = round(req_wave.min.value) * u.Angstrom
+            wmax_round = round(req_wave.max.value) * u.Angstrom
+
+            req_wave = (a.Wavelength(wmin_round, wmax_round))
```

Comment:
```suggestion
            wmin_round = round(req_wave.min.value) * u.Angstrom
            wmax_round = round(req_wave.max.value) * u.Angstrom
            req_wave = a.Wavelength(wmin_round, wmax_round)
```

Reply:
How does this logic account for different wavelength units?

Reply:
Where does round come from?

Reply:
its actually an inbuilt function 
https://www.geeksforgeeks.org/round-function-python/

Reply:
Huh, TIL